File: Java/src/main/java/minimal/model/Recipe.java
package minimal.model;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;

/**
 * Recipe for a meal. A recipe contains the list of the names of the needed ingredients and a
 * description how to prepare them.
 */
public class Recipe {

    private final Name nameOfMeal;
    private final List<Name> ingredients;

    public Recipe(Name nameOfMeal, Name... ingredients) {
        this.nameOfMeal = nameOfMeal;
        this.ingredients = Arrays.asList(ingredients);
    }

    public Name nameOfMeal() {
        return nameOfMeal;
    }

    public List<Name> listIngredients() {
        return Collections.unmodifiableList(ingredients);
    }

    public boolean canBePreparedWith(List<Name> availableIngredients) {
        return availableIngredients.containsAll(ingredients);
    }

    @Override
    public String toString() {
        return "Recipe to prepare " + nameOfMeal.toString();
    }

}

File: PHP/src/Model/Recipe.php
<?php

namespace Minimal\Model;

/**
 * Recipe for a meal. A recipe contains the list of the names of the needed ingredients and a
 * description how to prepare them.
 */
class Recipe
{
    private Name $nameOfMeal;
    private array $ingredients;

    public function __construct(Name $nameOfMeal, Name ...$ingredients)
    {
        $this->nameOfMeal = $nameOfMeal;
        $this->ingredients = $ingredients;
    }

    public function nameOfMeal(): Name
    {
        return $this->nameOfMeal;
    }

    public function listIngredients(): array
    {
        return $this->ingredients;
    }

    public function canBePreparedWith(array $availableIngredients): bool
    {
        return count(array_diff($this->ingredients, $availableIngredients)) === 0;
    }

    public function __toString()
    {
        return "Recipe to prepare " . $this->nameOfMeal->__toString();
    }
}
